src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -178,11 +198,11 @@ class OrderedRVD(
 
   override def coalesce(maxPartitions: Int, shuffle: Boolean): OrderedRVD = {
     require(maxPartitions > 0, "cannot coalesce to nPartitions <= 0")
-    val n = rdd.partitions.length
+    val n = crdd.partitions.length
     if (!shuffle && maxPartitions >= n)
       return this
     if (shuffle) {
-      val shuffled = crdd.coalesce(maxPartitions, shuffle = true)
+      val shuffled = stably(_.shuffleCoalesce(maxPartitions))
```

Comment:
So I still think this old code is wrong and we don't need to be shuffling twice here.  Can't we just compute the key ranges using the requested number of partitions and then call OrderedRVD.shuffle with the new ranges?

Reply:
@tpoterba @patrick-schultz  thoughts?

Reply:
it depends what you're doing. If you want to repartition after an expensive filter that removed 99.9% of your data, then shuffling twice is way better than scanning.